similaritymeasures/similaritymeasures.py

```diff
@@ -879,3 +879,111 @@ def dtw_path(d):
+    ----------
+    exp_data : array_like
+        Curve from your experimental data. exp_data is of (M, N) shape, where
+        M is the number of data points, and N is the number of dimmensions
+    num_data : array_like
+        Curve from your numerical data. num_data is of (P, N) shape, where P
+        is the number of data points, and N is the number of dimmensions
+    p: float,
+        1 <= p <= infinity. Which Minkowski p-norm to use.
+    threshold : positive int
+        If M * N * K > threshold, algorithm uses a Python loop instead of large temporary arrays.
+
+    Returns
+    -------
+    r : float
+        MSE."""
+    c = distance_matrix(exp_data, num_data, p=p, threshold=threshold)
+    return np.mean(np.square(np.diag(c)))def mae(exp_data, num_data, p=2, threshold=1000000):
+    """
+    Compute the Mean Absolute distance.
+    This computes the mean of absolute values of the distances 
+    between 2 corresponding points on the curve. This can use all 
+    distance metrics that are available in scipy.spatial.distance.cdist.
+
+    Parameters
+    ----------
+    exp_data : array_like
+        Curve from your experimental data. exp_data is of (M, N) shape, where
+        M is the number of data points, and N is the number of dimmensions
+    num_data : array_like
+        Curve from your numerical data. num_data is of (P, N) shape, where P
+        is the number of data points, and N is the number of dimmensions
+    p: float,
+        1 <= p <= infinity. Which Minkowski p-norm to use.
+    threshold : positive int
+        If M * N * K > threshold, algorithm uses a Python loop instead of large temporary arrays.
+
+    Returns
+    -------
+    r : float
+        MAE.
+    """
+    c = distance_matrix(exp_data, num_data, p=p, threshold=threshold)
+    return np.mean(np.diag(c))  # All values in c are positive since Minkowski distance is always positive
+
+def mse(exp_data, num_data, p=2, threshold=1000000):
+    """
+    Compute the Mean Squared distance.
+    This computes the mean of Sqaured values of the distances 
+    between 2 corresponding points on the curve. This can use all 
+    distance metrics that are available in scipy.spatial.distance.cdist.
+
+    Parameters
+    ----------
+    exp_data : array_like
+        Curve from your experimental data. exp_data is of (M, N) shape, where
+        M is the number of data points, and N is the number of dimmensions
+    num_data : array_like
+        Curve from your numerical data. num_data is of (P, N) shape, where P
+        is the number of data points, and N is the number of dimmensions
+    p: float,
+        1 <= p <= infinity. Which Minkowski p-norm to use.
+    threshold : positive int
+        If M * N * K > threshold, algorithm uses a Python loop instead of large temporary arrays.
+
+    Returns
+    -------
+    r : float
+        MSE."""
+    c = distance_matrix(exp_data, num_data, p=p, threshold=threshold)
+    return np.mean(np.square(np.diag(c)))
```

Comment:
Like before, replace the calls here such that the function is `O(n)` complexity, instead of `O(n*n)`.

```python
c = np.square(exp_data - num_data)
return np.mean(c)
```